From 99a03bc528ed4ade5631e39d02b11bd7c74ea630 Mon Sep 17 00:00:00 2001
From: =?utf-8?q?J=C3=B6rg=20Sommer?= <joerg@alea.gnuu.de>
Date: Thu, 21 Aug 2008 11:34:48 +0200
Subject: [PATCH 2/5] Replace string operations by shell expressions
MIME-Version: 1.0
Content-Type: text/plain; charset=utf-8
Content-Transfer-Encoding: 8bit

The task to extract these fields from the string can be done with shell
utils which saves pipes, forkes, …
---
 getbuildlog |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/getbuildlog b/getbuildlog
index daa0910..7b4a781 100755
--- a/getbuildlog
+++ b/getbuildlog
@@ -85,8 +85,10 @@ getbuildlog() {
     sed -i -e "s/%2B/\+/g" -e "s/%3A/:/g" $ALL_LOGS
 
     for match in `grep -E -o "$PATTERN" $ALL_LOGS`; do
-        ver=`echo $match | cut -d'&' -f3 | cut -d'=' -f2`
-        arch=`echo $match | cut -d'&' -f4 | cut -d'=' -f2`
+        ver=${match##*ver=}
+        ver=${ver%%&*}
+        arch=${match##*arch=}
+        arch=${arch%%&*}
 	match=`echo $match | sed -e 's/\+/%2B/g'`
         wget -O "${PACKAGE}_${ver}_${arch}.log" "$BASE/$match&file=log"
     done
-- 
1.5.6.3

